### PR TITLE
feat: collect jira status

### LINKED ISSUE #2181: [Feature][ticket] add standard status in changelogs
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Now we only stored original status in changelogs entity(in 'status' type), to calculate some metrics (eg: #1796  completion_date), we need to know changelogs of standard status in changelogs entity.  
```bash
postgres@postgres:lake> select * from changelogs where field_name = 'status' limit 1;
"id","created_at","updated_at","_raw_data_params","_raw_data_table","_raw_data_id","_raw_data_remark","issue_id","author_id","author_name","field_id","field_name","from_value","to_value","created_date"
"jira:JiraChangelogItem:1:118864:status","2022-05-30 08:59:53.557919+00","2022-05-30 08:59:53.557919+00","{""ConnectionId"":1,""BoardId"":68}","_raw_jira_api_changelogs","60805","","jira:JiraIssue:1:15144","5ecfbd0984083c0c12e5af8f","lin.hao","","status","已关闭","重新打开","2021-06-21 12:00:52.757+00"
```

### Use case

To calculate completion_date of issues, we need to know the latest time when issue come into a done status. 

### Related issues

#1796 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by narrowizard**
# Summary

### Does this close any open issues?
closes #2181 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/jira/models/connection.go [hunk 1]
```diff
@@ -45,19 +45,27 @@ type JiraConnection struct {
...
+	return "_tool_jira_connections"
+}
+
 type JiraIssueTypeMapping struct {
 	ConnectionID uint64 `gorm:"primaryKey" json:"jiraConnectionId" validate:"required"`
 	UserType     string `gorm:"type:varchar(50);primaryKey" json:"userType" validate:"required"`
 	StandardType string `gorm:"type:varchar(50)" json:"standardType" validate:"required"`
 }
 
+func (JiraIssueTypeMapping) TableName() string {
```
Comment (mindlesscloud) [inline plugins/jira/models/connection.go:58]: do we need this?

File: plugins/jira/models/connection.go [hunk 2]
```diff
@@ -45,19 +45,27 @@ type JiraConnection struct {
...
 type JiraIssueStatusMapping struct {
 	ConnectionID   uint64 `gorm:"primaryKey" json:"jiraConnectionId" validate:"required"`
 	UserType       string `gorm:"type:varchar(50);primaryKey" json:"userType" validate:"required"`
 	UserStatus     string `gorm:"type:varchar(50);primaryKey" json:"userStatus" validate:"required"`
 	StandardStatus string `gorm:"type:varchar(50)" json:"standardStatus" validate:"required"`
 }
 
-func (JiraConnection) TableName() string {
-	return "_tool_jira_connections"
+func (JiraIssueStatusMapping) TableName() string {
```
Comment (mindlesscloud) [inline plugins/jira/models/connection.go:69]: there is not a table named `_tool_jira_issue_status_mappings`

File: plugins/jira/models/connection.go (hunk 1 above)
Comment (narrowizard) [reply plugins/jira/models/connection.go:58]: It reports an error that cannot find table `jira_issue_type_mappings` when running issue extractor.

File: models/migrationscripts/updateSchemas2022061402.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	"context"
+
+	"github.com/apache/incubator-devlake/models/domainlayer/ticket"
+	"gorm.io/gorm"
+)
+
+type updateSchemas2022061402 struct{}
+
+func (*updateSchemas2022061402) Up(ctx context.Context, db *gorm.DB) error {
+	return db.Migrator().AutoMigrate(&ticket.Changelog{})
```
[the remark this answers is not in the document]
Comment (narrowizard) [reply models/migrationscripts/updateSchemas2022061402.go]: fixed, but still refer to common package's structure. Is that allowed?

File: plugins/jira/tasks/shared.go [hunk 4]
```diff
@@ -35,3 +38,26 @@ func GetTotalPagesFromResponse(res *http.Response, args *helper.ApiCollectorArgs
...
+	} else if statusKey == "new" {
+		return ticket.TODO
+	} else {
+		return ticket.IN_PROGRESS
+	}
+}
+
+func GetStatusInfo(db *gorm.DB) (map[string]models.JiraStatus, error) {
+	data := make([]models.JiraStatus, 0)
+	err := db.Table("_tool_jira_statuses").Scan(&data).Error
```
[the remark this answers is not in the document]
Comment (narrowizard) [reply plugins/jira/tasks/shared.go]: fixed.

File: plugins/jira/models/connection.go (hunk 2 above)
Comment (narrowizard) [reply plugins/jira/models/connection.go:69]: Seems used by issue_extractor? Is this a mistake?  
https://github.com/apache/incubator-devlake/blob/04e4a01d94c13a98b240628205f3ce3123f2a3ab/plugins/jira/tasks/issue_extractor.go#L61

File: plugins/jira/models/migrationscripts/updateSchemas20220614.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+	"github.com/apache/incubator-devlake/plugins/jira/models"
+	"gorm.io/gorm"
+)
+
+type UpdateSchemas20220614 struct{}
+
+func (*UpdateSchemas20220614) Up(ctx context.Context, db *gorm.DB) error {
+	return db.Migrator().AutoMigrate(
+		&models.JiraStatus{},
```
[the remark this answers is not in the document]
Comment (narrowizard) [reply plugins/jira/models/migrationscripts/updateSchemas20220614.go]: fixed, but still refer to common package's structure. Is that allowed?

File: plugins/jira/models/migrationscripts/updateSchemas20220614.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"context"
+
+	"github.com/apache/incubator-devlake/models/common"
+	"gorm.io/gorm"
+)
+
+type JiraStatus struct {
+	common.NoPKModel
```
Comment (mindlesscloud) [inline plugins/jira/models/migrationscripts/updateSchemas20220614.go:28]: we are supposed to import the `NoPKModel` in the package `github.com/apache/incubator-devlake/models/migrationscripts/archived`

File: models/migrationscripts/updateSchemas2022061402.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"context"
+	"time"
+
+	"github.com/apache/incubator-devlake/models/domainlayer"
+	"gorm.io/gorm"
+)
+
+type Changelog struct {
+	domainlayer.DomainEntity
```
Comment (mindlesscloud) [inline models/migrationscripts/updateSchemas2022061402.go:29]: we are supposed to import `DomainEntity` from `github.com/apache/incubator-devlake/models/migrationscripts/archived`

File: plugins/jira/models/migrationscripts/updateSchemas20220614.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (mindlesscloud) [reply plugins/jira/models/migrationscripts/updateSchemas20220614.go]: No, we should import from `github.com/apache/incubator-devlake/models/migrationscripts/archived`

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**